Feature/select db to connect on

gplssm opened the pull request:
This pull request contains an enhancement to select section of config.ini in order to connect to different databases. To use this feature type `conn = db.connection(db_section='oedb')` to obtain parameters for database configured in section "oedb".

Commit:
Revert "added translations for powerplants from database (e.g. mineral oil...)"

This reverts commit d0000d5a9e438bbfa2c68a0303402c0da7aa6fdb.

Commit:
Revert "Revert "added translations for powerplants from database (e.g. mineral oil...)""

This reverts commit da195dd8cd326a6645e110e6d07f243d26074789.

Commit:
database selection by keyword argument

uvchik commented:
Thank you, could you add a docstring, that explains the new parameter?

Commit:
docstring added

gplssm commented:
Done. Ready to merge?

uvchik commented:
Thanks for you work but as we use sphinx with numpydoc the docstring should have the following form (https://github.com/numpy/numpy/blob/master/doc/example.py):
```
    r"""A one-line summary that does not use variable names or the
    function name.

    Several sentences providing an extended description. Refer to
    variables using back-ticks, e.g. `var`.

    Parameters
    ----------
    var1 : array_like
        Explaining the variable. Array_like means all those objects
-- lists, nested lists, etc. --
        that can be converted to an array.  We can also refer to
        variables like `var1`.
    var2 : int
        The type above can either refer to an actual Python type
        (e.g. ``int``), or describe the type of the variable in more
        detail, e.g. ``(N,) ndarray`` or ``array_like``.
    
    Returns
    -------
    type
        Explanation of anonymous return value of type ``type``.

```
Could you also add a docstring for the connection function just one sentence and a link to the engine-function. I think that should look like this ```:py:func:`engine` ``` (http://www.sphinx-doc.org/en/stable/domains.html#role-py:func).

Commit:
docstring revised

...according to given numpy example

gplssm commented:
Thanks for your time explaining things to a newbie! ;) Result see 86addf05d1f6e48472059b5ec9b11f0e531ef99d

Can I find a explanation of the config.ini file somewhere? I would like to link this in the docstring.

I linked the numpy example docstring example to the oemof documentation to avoid explaining it to every oemof-newbie.

Btw, do the oemof devs have a standard procedure to test doc strings locally? Is it described somewhere?

uvchik commented:
Thank you very much, that is really a big improvement.

> Can I find a explanation of the config.ini file somewhere?

There is something in the docstring of the [config.py-file](../blob/dev/oemof/db/config.py).

I use the following code to compile a documentation:
```sh
sphinx-apidoc -o .../git_home/oemof.db/doc/api /.../git_home/oemof.db/oemof
sphinx-build -b html .../git_home/oemof.db/doc .../git_home/oemof.db/doc/_build/html
```

Commit:
Added reference to config.py

uvchik committed:
pep8 fixes

uvchik commented:
I fixed some pep8 issues. I think you can merge it.

gplssm commented:
Thanks for cleaning up! I'll try to avoid its necessity in the future ;)

@gnn you do the merge? Or why have you assigned yourself to it?

gnn commented:
Yeah, I assigned myself because I wanted to look at it, mainly to see how it's done as I wanted to implement this too. ;) So many thanks for doing it, as I stumbled over this missing feature too.
Is it ok if I rename `db_section` to just `section`? We're in the `db` module anyway so it should be clear that we're dealing with the database anyway and I'd try to avoid repetition where possible. But that's purely cosmetic and if it breaks too much of your existing code, I'm going to leave it be.

There are a two other things I found which I'm going to fix in the meantime:

  - wrong return type in `engine` docstring,
  - I'm pretty sure the exception handling in `engine` doesn't work as intended.

gnn committed:
Fix return type in `engine` docstring

gnn committed:
Fix `config.ini` documentation reference

I guess there shouldn't have been a space between ":download:" and
"`config.py`" and that the `config` module was meant so I wanted to
reference that. Instead I ended up linking to the `README`, as the
structure and location of `congig.ini` is described pretty clearly
there. In order to do that I had to add an explicit [label][0] to
the `Configuration` section of the `README` as I couldn't get
hyperlinking to work properly otherwise.

[0]: http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-arbitrary-locations

gnn committed:
Rephrase note on `config.ini`

gnn committed:
Update README to reflect new feature

gnn committed:
Mention new feature in "What's New"

gnn commented:
I changed some documentation stuff. Will have a look at the exception handling later. I also added an entry in the `What's New` section and you can add yourself as a contributor there, @gplssm . I didn't know whether you wanted to have your realname there or an alias. Your choice.

gnn commented:
I checked out the exception handling and testing it takes a little longer than I expected so I'm opening an issue for it so I can do it later. I was written by me anyway and isn't particular to this branch. It only showed up in the commits because it was touched by the PEP8 fixes, so you're not the only one making those necessary @gplssm ;).
All that's left now is my question about parameter renaming. If that's OK with you, just say so and I'll do it and proceed with the merge. If not, you can just go ahead and merge.

gplssm committed:
Add contributer to "What's new"

gplssm commented:
Great work on the details. I really appreciate!
Renaming `db_section` to `section` is totally fine for me and it makes sense to avoid double information.
Thanks for the investigation of the testing. Go ahead for the merge @gnn 

Btw, it's a good workflow that you assigned yourself to the pull request for checking new feature, its implementation and finally doing the merge.

gnn committed:
Rename optional parameter
